mr_project/mr_app/views.py

```diff
@@ -1,9 +1,231 @@
+import json
+
 from django.shortcuts import render
 
 # Create your views here.
+from time import time
+from django.http import HttpResponse, HttpResponseBadRequest, HttpResponseNotFound, JsonResponse
+import neo4j
+from django.views.decorators.csrf import csrf_exempt
+from requests import codes
+from . import DRIVER, EMOTIONS, USERS
+from datetime import datetime
+
+#WRITE TRANSACTION FUNCTIONS
+def create_user_tx(tx, name):
+    result = tx.run('CREATE (u:User{name:$name}) RETURN u.name AS username', name=name)
+    record = result.single()
+    result.consume()
+    return record['username']
+
+
+def create_emotion_tx(tx, name):
+    result = tx.run('CREATE (em:Emotion {name:$name})  RETURN em.name AS emotion_label', name=name)
+    record = result.single()
+    result.consume()
+    return record['emotion_label']
+
+
+def create_epoch_relationship_between_user_and_emotion(tx, emotion, user, epoch_timestamp):
+    result = tx.run(f'MATCH (em:Emotion {{name:"{emotion}"}}), (u:User {{name:"{user}"}}) CREATE (u)-[ts:Timestamp{{epoch:"{epoch_timestamp}"}}]->(em) RETURN ts.epoch as seconds')
+    record = result.single()
+    result.consume()
+    return record['seconds']
```

Comment:
Are these lines necessary if we don't need the return value?

Reply:
yep